pkg/volumes/do/volumes.go

```diff
@@ -136,54 +208,146 @@ func (a *DOVolumes) findVolumes(filterByRegion bool) ([]*volumes.Volume, error)
-				EtcdName:  a.ClusterName + "-" + clusterKey,
-			}
+		glog.V(2).Infof("Check for matching droplet tag - doTag=%s for droplet name = %s", strings.ToUpper(doTag), a.dropletName)
+
+		// Also check if this tag is seen in the droplet.
+		dropletTagfound := a.Contains(a.dropletTags, doTag)
+		if !dropletTagfound {
+			return false
+		} else {
+			glog.V(2).Infof("Matching tag found for doTag=%s in droplet name = %s", strings.ToUpper(doTag), a.dropletName)
+		}
+	}
 
-			if len(doVolume.DropletIDs) == 1 {
-				vol.AttachedTo = strconv.Itoa(doVolume.DropletIDs[0])
-				vol.LocalDevice = getLocalDeviceName(&doVolume)
+	return true
+}
+
+func (a *DOVolumes) matchesTags(volume *godo.Volume) bool {
+
+	tagFound := a.matchDropletTags(volume)
+
+	// Check if match is Found, otherwise, just return if nothing found.
+	if !tagFound {
+		return false
+	}
+
+	matchingTagCount := 0
+	// now that the tags are matched, check tagkeys if it exists in both volume and droplet tags, and ensure the key values in the volume and droplet tags match.
+	for _, matchTag := range a.matchTagKeys {
+		for _, volumeTag := range volume.Tags {
+			vt := strings.SplitN(volumeTag, ":", -1)
+			if len(vt) < 1 {
+				// not interested in this tag.
+				continue
 			}
 
-			glog.V(2).Infof("Matching DO Volume found with name=%s ID=%s etcdname=%s mountname=%s", doVolume.Name, doVolume.ID, vol.EtcdName, vol.MountName)
+			vtKey := vt[0]
+			vtValue := vt[1]
+
+			glog.V(2).Infof("Matching tag keys for vtKey=%s; vtValue = %s; matchTagKey = %s", strings.ToUpper(vtKey), strings.ToUpper(vtValue), strings.ToUpper(matchTag))
+
+			if strings.ToUpper(matchTag) == strings.ToUpper(vtKey) {
+				glog.V(2).Infof("Match found for tag keys for vtKey=%s; vtValue = %s; matchTagKey = %s", strings.ToUpper(vtKey), strings.ToUpper(vtValue), strings.ToUpper(matchTag))
+
+				// match found, verify if this tag exists in dropletTags and find a matching value.
+				for _, dropletTag := range a.dropletTags {
+					dt := strings.SplitN(dropletTag, ":", -1)
+					dtKey := dt[0]
+					dtValue := dt[1]
+
+					glog.V(2).Infof("Matching droplet tag keys for dtKey=%s; dtValue = %s; matchTagKey = %s", strings.ToUpper(dtKey), strings.ToUpper(dtValue), strings.ToUpper(matchTag))
+
+					if strings.ToUpper(matchTag) == strings.ToUpper(dtKey) {
+						glog.V(2).Infof("Matching droplet key FOUND for dtKey=%s; dtValue = %s; matchTagKey = %s", strings.ToUpper(dtKey), strings.ToUpper(dtValue), strings.ToUpper(matchTag))
+
+						// droplet tag also matched, check if the value matches.
+						if strings.ToUpper(dtValue) == strings.ToUpper(vtValue) {
+							glog.V(2).Infof("Everything matched for matchTagKey = %s", strings.ToUpper(matchTag))
+
+							matchingTagCount++
+						}
+					}
+				}
+			}
 
-			myvolumes = append(myvolumes, vol)
 		}
+	}
 
+	if len(a.matchTagKeys) == matchingTagCount {
+		return true
 	}
 
-	return myvolumes, nil
+	return false
+}
+
+// Contains tells whether a contains x.
```

Comment:
s/a/tags/

Reply:
I didn't get this comment.

Reply:
Clarified on Slack that this was about renaming `a` to `tags` to have the GoDoc match the code.